crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,291 @@
+//! Describe a subset of the genesis manifest in easily human readable format
+
+use std::collections::HashMap;
+
+use miden_lib::{
+    AuthScheme,
+    account::{auth::RpoFalcon512, faucets::BasicFungibleFaucet, wallets::create_basic_wallet},
+};
+use miden_node_utils::crypto::get_rpo_random_coin;
+use miden_objects::{
+    Felt, FieldElement, StarkField, Word,
+    account::{
+        Account, AccountBuilder, AccountDelta, AccountFile, AccountId, AccountStorageDelta,
+        AccountStorageMode, AccountType, AccountVaultDelta, AuthSecretKey, FungibleAssetDelta,
+        NonFungibleAssetDelta,
+    },
+    asset::{FungibleAsset, TokenSymbol},
+    crypto::dsa::rpo_falcon512::SecretKey,
+};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+
+use crate::GenesisState;
+
+mod errors;
+use self::errors::Error;
+
+#[cfg(test)]
+mod tests;
+
+/// Represents an account, either a wallet or a faucet
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum AccountConfig {
+    Wallet(WalletConfig),
+    Faucet(FaucetConfig),
+}
+
+/// `false` doesn't pass the `syn::Path` parsing, so we do one level indirection
+const fn ja() -> bool {
+    true
+}
+
+/// Represents a wallet, containing a set of assets
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct WalletConfig {
+    /// Provide a name, that will be used for the keyfile.
+    #[serde(default)]
+    name: Option<String>,
```

Comment:
I'm in two minds about this. Its a nice touch but do we really need it?

Should we not just generate files `account_{i}` in the order they're specified? Though I notice this won't be true as they're separate properties 🤔

Reply:
Removed the `name` fields